app/src/main/java/com/codesoom/assignment/application/UserService.java

```diff
@@ -0,0 +1,47 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.dto.UserData;
+import com.codesoom.assignment.exception.UserNotFoundException;
+import com.codesoom.assignment.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public User getUser(Long id) {
+        return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+    }
+
+    public User createUser(UserData userData) {
+        User user = User.builder()
+                .name(userData.getName())
+                .password(userData.getPassword())
+                .email(userData.getEmail())
+                .build();
+
+        return userRepository.save(user);
+    }
+
+    public User updateUser(Long targetId, UserData source) {
+        User user = getUser(targetId);
+
+        user.change(source.getName(),
+                source.getPassword(),
+                source.getEmail());
+
+        return userRepository.save(user);
+    }
+
+    public void deleteUser(Long targetId) {
+        User user = getUser(targetId);
+
+        userRepository.delete(user);
```

Comment:
만약 시간이 넉넉하다면 이런 비슷한 모양이 나도록 한번 수정해 보세요.
```suggestion
        userRepository.deleteByUserId(targetId);
```

Reply:
네! id 값으로 삭제하도록 수정하였으며,  기존에 있었던 exception 처리를 하기 위해 try-catch 문을 추가하였습니다.